code/LeftAndMain.php

```diff
@@ -1223,24 +1223,7 @@ public function getEditForm($id = null, $fields = null)
             $form->addExtraClass('cms-previewable');
         }
         $form->addExtraClass('fill-height');
-
-        if ($record->hasMethod('getCMSCompositeValidator')) {
-            // As of framework v4.7, a CompositeValidator is always available form a DataObject, but it may be
-            // empty (which is fine)
-            $form->setValidator($record->getCMSCompositeValidator());
-        } elseif ($record->hasMethod('getCMSValidator')) {
-            // BC support for framework < v4.7
-            $validator = $record->getCMSValidator();
-
-            // The clientside (mainly LeftAndMain*.js) rely on ajax responses
-            // which can be evaluated as javascript, hence we need
-            // to override any global changes to the validation handler.
-            if ($validator) {
-                $form->setValidator($validator);
-            }
```

Comment:
All of this boils down to `$form->setValidator($record->getCMSCompositeValidator())`, because all `DataObject` classes have that method, and `$record` will always be a `DataObject`.